### PR TITLE
[DEVX-1824] Bump cypress to 10.3.1

### PR DESCRIPTION
**Pull request by tianfeng92**

### CODE REVIEW

File: tests/kitchen-sink-tests/.sauce/config_on_sauce_mac.yml [hunk 1]
```diff
@@ -17,11 +17,21 @@ cypress:
...
+    platformName: "macOS 12"
     browser: "chrome"
-    browserVersion: "98"
+    shard: concurrency
     config:
       testingType: "e2e"
       specPattern: [ "**/*cy.js" ]
       excludeSpecPattern: ["cypress/e2e/failing/todo.cy.js"]
+
+# commented mac + firefox test for failed to connect to the browser
```
Comment (alexplischke) [inline tests/kitchen-sink-tests/.sauce/config_on_sauce_mac.yml:29]: Does this constantly fail though? If you run it with retry, would it pass?

Comment (tianfeng92) [reply tests/kitchen-sink-tests/.sauce/config_on_sauce_mac.yml:29]: It passed locally but failed constantly twice on the pipeline. 🤔

Comment (tianfeng92) [reply tests/kitchen-sink-tests/.sauce/config_on_sauce_mac.yml:29]: failed constantly for 5 times 😂

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -135,14 +150,31 @@ jobs:
...
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+
+      - name: Use Node.js v14.x
+        uses: actions/setup-node@v1
```
[the remark this answers is not in the document]
Comment (tianfeng92) [reply .github/workflows/test.yml:164]: sure 👍

[the remark this answers is not in the document]
Comment (tianfeng92) [reply .github/workflows/test.yml]: that sounds very like「改正gaizheng」😆

**Review by alexplischke — APPROVED** [review]

**Pull request merged**